packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -903,21 +902,30 @@ class ServiceExtensionManager extends Disposer {
   }
 
   Future<void> _onMainIsolateChanged() async {
-    if (_mainIsolate.value == null) {
+    if (_isolateManager.mainIsolate.value == null) {
       _mainIsolateClosed();
       return;
     }
     _checkForFirstFrameStarted = false;
 
-    final isolateRef = _mainIsolate.value;
-    final Isolate isolate = await _service.getIsolate(isolateRef.id);
-    if (isolateRef != _mainIsolate.value) {
+    final isolateRef = _isolateManager.mainIsolate.value;
+    final Isolate isolate = await _isolateManager.getIsolateCached(isolateRef);
+
+    await _registerIsolate(isolate, isolateRef);
+  }
+
+  Future<void> _registerIsolate(Isolate isolate,
```

Comment:
parameter name should not be previousIsolateRef. It should be expectedMainIsolateRef or similar.